heron/schedulers/tests/java/com/twitter/heron/scheduler/nomad/NomadSchedulerTest.java

```diff
@@ -0,0 +1,366 @@
+    PackingPlan validPlan =
+        new PackingPlan(PACKING_PLAN_ID, containers);
+
+    // Fail to submit due to client failure
+    Job[] jobs = {new Job(), new Job(), new Job()};
+    List<Job> jobList = Arrays.asList(jobs);
+    Mockito.doReturn(jobList).when(scheduler).getJobs(validPlan);
+    PowerMockito.doThrow(new RuntimeException()).when(NomadScheduler.class,
+        "startJobs", Mockito.any(NomadApiClient.class), anyVararg());
+    Assert.assertFalse(scheduler.onSchedule(validPlan));
+
+    // succeed
+    Mockito.doReturn(jobList).when(scheduler).getJobs(validPlan);
+    PowerMockito.doNothing().when(NomadScheduler.class, "startJobs",
+        Mockito.any(NomadApiClient.class), anyVararg());
+    Assert.assertTrue(scheduler.onSchedule(validPlan));
+  }
+
+  @Test
+  public void testOnRestart() throws Exception {
+    PowerMockito.mockStatic(NomadScheduler.class);
+
+    // Construct RestartTopologyRequest to restart container
+    Scheduler.RestartTopologyRequest restartTopologyRequest =
+        Scheduler.RestartTopologyRequest.newBuilder()
+            .setTopologyName(TOPOLOGY_NAME)
+            .setContainerIndex(CONTAINER_INDEX)
+            .build();
+
+    scheduler.initialize(this.mockConfig, this.mockRuntime);
+
+    // Fail to restart
+    PowerMockito.when(NomadScheduler.getTopologyContainerJob(
+        Mockito.any(NomadApiClient.class),
+        Mockito.anyString(), Mockito.anyInt())).thenReturn(new Job());
+    PowerMockito.doThrow(new RuntimeException()).when(
+        NomadScheduler.class, "restartJobs",
+        Mockito.any(NomadApiClient.class), Mockito.any(Job.class));
+    Assert.assertFalse(scheduler.onRestart(restartTopologyRequest));
+
+    // succeed to restart
+    PowerMockito.when(NomadScheduler.getTopologyContainerJob(
+        Mockito.any(NomadApiClient.class),
+        Mockito.anyString(), Mockito.anyInt())).thenReturn(new Job());
+    PowerMockito.doNothing().when(NomadScheduler.class, "restartJobs",
+        Mockito.any(NomadApiClient.class), Mockito.any(Job.class));
+    Assert.assertTrue(scheduler.onRestart(restartTopologyRequest));
+
+    // Construct RestartTopologyRequest to restart whole topology
+    restartTopologyRequest =
+        Scheduler.RestartTopologyRequest.newBuilder()
+            .setTopologyName(TOPOLOGY_NAME)
+            .setContainerIndex(-1)
+            .build();
+
+    scheduler.initialize(this.mockConfig, this.mockRuntime);
+
+    Job[] jobs = {new Job(), new Job(), new Job()};
+    List<Job> jobList = Arrays.asList(jobs);
+
+    // Fail to restart
+    PowerMockito.when(NomadScheduler.getTopologyJobs(Mockito.any(NomadApiClient.class),
+        Mockito.anyString())).thenReturn(jobList);
+    PowerMockito.doThrow(new RuntimeException()).when(NomadScheduler.class,
+        "restartJobs", Mockito.any(), anyVararg());
+    Assert.assertFalse(scheduler.onRestart(restartTopologyRequest));
+
+    // succeed to restart
```

Comment:
nit: s/succeed/Succeed

There are a few of them in this file

Reply:
will fix